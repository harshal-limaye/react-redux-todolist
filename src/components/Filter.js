import React from 'react';

class Filter extends React.Component {
    render() {
        return (
            <ul className="filters">
                <li className={this.getClass('ALL')} onClick={() => this.setFilter('ALL')}>All</li><li>/</li>
                <li className={this.getClass('ACTIVE')} onClick={() => this.setFilter('ACTIVE')}>Active</li><li>/</li>
                <li className={this.getClass('COMPLETED')} onClick={() => this.setFilter('COMPLETED')}>Completed</li>
            </ul>
        );
    }

    getClass = (filter) => this.props.visibility === filter ? 'active' : '';

    setFilter = (filter) => this.props.setFilter(filter);
}

export default Filter;