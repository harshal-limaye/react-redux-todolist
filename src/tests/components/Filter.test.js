import React from 'react';
import { shallow } from 'enzyme';
import Filter from '../../components/Filter';

let wrapper;

describe('should test Filter component', () => {
    beforeEach(() => {
        const visibility = 'ALL';
        wrapper = shallow(<Filter visibility={visibility} />);
    });

    test('should render component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should call getClass method', () => {
        const className = wrapper.instance().getClass('ALL');
        expect(className).toBe('active');
    });

    test('should call setFilter method', () => {
        const visibility = 'ALL';
        const setFilter = jest.fn();
        const component = shallow(<Filter visibility={visibility} setFilter={setFilter} />);
        component.instance().setFilter('ACTIVE');
        expect(setFilter).toBeCalled();
    });
});