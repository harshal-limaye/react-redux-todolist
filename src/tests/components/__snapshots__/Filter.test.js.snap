// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Filter component should render component correctly 1`] = `
<ul
  className="filters"
>
  <li
    className="active"
    onClick={[Function]}
  >
    All
  </li>
  <li>
    /
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Active
  </li>
  <li>
    /
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Completed
  </li>
</ul>
`;
